odl/trafos/wavelet.py

```diff
@@ -750,11 +828,26 @@ def __init__(self, ran, nscales, wbasis, mode):
 
             'per': periodization -- like periodic-padding but gives the
             smallest possible number of decomposition coefficients.
+
+            With J.-O. Stomberg's biorthogonal wavelets only signal
+            extension mode 'sym' is available.
         """
         self.nscales = int(nscales)
-        self.wbasis = wbasis
         self.mode = str(mode).lower()
 
+        if isinstance(wbasis, pywt.Wavelet):
+            self.wbasis = wbasis
+        else:
+            try:
+                wbasis = wbasis + ''
+                if wbasis.startswith('JOS'):
+                    self.wbasis = wbasis
```

Comment:
I suppose you refer mainly to ` wbasis = wbasis + ''`
This was a fix suggested by @kohr-h, previously I had
 
 `if isinstance(wbasis, pywt.Wavelet):`
            `self.wbasis = wbasis`
        `elif isinstance(wbasis, str):`
            `if wbasis[0:6] == 'biorth':`
                `self.wbasis = wbasis`
            `else:`
                `self.wbasis = pywt.Wavelet(wbasis)`

but this did not work (and would have certainly needed some improve). 

As you said `self.wbasis` can be either a `str` or a `pywt.Wavelet`. When it is a `str` one needs to distinguish if it refers to the wavelet families within pyWavelets (quite a many combinations with all possible filters in all wavelet families)  of to the Jan-Olov's wavelets. If it refers to Jan-Olov's then it will further be passed as a `str` else transformed to `pywt.Wavelet`. As I said my previous attempt passed all `str` to `BiorthWaveletTranform` even when they referred to some of pyWavelets wavelet families...

Would appreciate a comment how to better formulate this.